src/views/ajax.js

```diff
@@ -5,6 +5,36 @@ const arrowIcon = require('./open').arrowIcon;
 const icons = require('../assets/icons').default;
 const statusIcon = require('../assets/icons').statusIcon;
 
+/**
+ * removeOrigin - function to remove origin from a URL.
+ *
+ * @param {String} url URL to remove the origin from.
+ * @param {String} origin Origin to remove from URL.
+ */
+const removeOrigin = (url = '', origin = window.location.origin) => {
+  if (url.substr(0, origin.length) === origin) {
+      url = url.substr(origin.length, url.length);
+  }
+
+  return url;
+};
+
+/**
+ * removeOriginFromUrl - function to remove origin from a URL
+ *                       reguarding `http` and `https`.
+ *
+ * @param {String} url URL to remove the origin from.
+ * @param {String} origin Origin to remove from URL.
+ */
+const removeOriginFromUrl = (url = '', origin = window.location.origin) => {
+  url = url.trim();
+
+  url = removeOrigin(url, origin.replace(/^http\:\/\//, 'https://'));
```

Comment:
I will add the comments. Basically, it covers all 4 cases for the `url` <-> `origin` pair:

| url protocol | origin protocol |
| --- | --- |
| http | http |
| http | https |
| https | http |
| https | https |

Thus if protocol is `http` we swap it to `https` and replace it (if it is not we leave it `https`).
Fo the second `removeOrigin` function call it is the same but for `https` - if the protocol is `https` we swap it to `http` and replace it (if it is not we leave it `http`).